R/lightgbm.R

```diff
@@ -260,6 +260,22 @@ sort_args <- function(args) {
   args
 }
 
+# in lightgbm <= 3.3.2, predict() for multiclass classification produced a single
+# vector of length num_observations * num_classes, in row-major order
+#
+# in versions after that release, lightgbm produces a numeric matrix with shape
+# [num_observations, num_classes]
+#
+# this function ensures that multiclass classification predictions are always
+# returned as a [num_observations, num_classes] matrix, regardless of lightgbm version
+reshape_lightgbm_multiclass_preds <- function(preds, num_rows) {
+    n_preds_per_case <- length(preds) / num_rows
+    if (is.vector(preds) && n_preds_per_case > 1) {
+        preds <- matrix(preds, ncol = npred_per_case, byrow = TRUE)
```

Comment:
I'm seeing:

```
══ Failed tests ════════════════════════════════════════════════════════════════
── Failure (test-lightgbm.R:141:3): boost_tree with lightgbm ───────────────────
`{ ... }` threw an unexpected error.
Message: object 'npred_per_case' not found
```

Could you change this line to: 
```suggestion
        preds <- matrix(preds, ncol = npreds_per_case, byrow = TRUE)
```